pkg/controller/storagecluster/storagecluster_controller_test.go

```diff
@@ -304,11 +304,12 @@ func TestThrottleStorageDevices(t *testing.T) {
 							StorageClassName: &fakestorageClassName,
 						},
 					},
-					Portable: true,
+					Portable:   true,
+					DeviceType: "hdd",
```

Comment:
I'd like to see cases of different possible values of DeviceType covered. Also, a case for invalid DeviceType.

Reply:
Right, but this is not the topic of this PR. That was introduced in a different PR. So we can (and should) enhance that in a separate PR.

Reply:
If this is not the topic of this PR, why does it add `DeviceType: "hdd"` to test case?
Honestly, I've lost track of "Tune disks" PRs and enhancements :sweat_smile:

Reply:
> If this is not the topic of this PR, why does it add `DeviceType: "hdd"` to test case?

This is exactly what I told @crombus: Using this to fix the test failure is wrong. It is just avoiding the code path.
Instead we need to make the fake cluster so that it has platform info, so the new code does not error out.

> Honestly, I've lost track of "Tune disks" PRs and enhancements sweat_smile

:wink: